server/download.go

```diff
@@ -474,6 +474,9 @@ func downloadBlob(ctx context.Context, opts downloadOpts) (cacheHit bool, _ erro
 	case errors.Is(err, os.ErrNotExist):
 	case err != nil:
 		return false, err
+	case len(opts.digest) <= 0:
```

Comment:
I think this is not the right place for this check. This is checking the error from looking up the file path. Instead we should check that the opts.Digest is not empty at the start of the function.

Something like this:
```
func downloadBlob(ctx context.Context, opts downloadOpts) (cacheHit bool, _ error) {
	if opts.digest == "" {
		return false, fmt.Errorf(("%s: %s"), opts.mp.GetNamespaceRepository(), "digest is is empty")
	}
```